src/ruby/puppetserver-lib/puppet/server/settings.rb

```diff
@@ -30,6 +30,22 @@ def generate_new_local_settings
                                                             push_settings_globally: false)
   end
 
+  # This method does the identical operation to generate_new_local_settings with
+  # the exception that it uses config passed in as a parameter rather than the value
+  # of @puppet_config.
+  #
+  # This function is used to create a compile context in which we can compile
+  # things for a bolt project. In that scenario we need to set all puppet settings
+  # such that puppet won't read things like puppet.conf, so we pass in a config
+  # seperate from the one used for normal puppet compilation.
+  def generate_standalone_local_settings(standalone_config)
```

Comment:
I think my thought process was mostly about being risk averse about these changes, so I created a new function that would only be used in the AST compiler right now and doesn't have any potential to create issues elsewhere. It's probably not particularly dangerous just to change the original function though, since I don't think anything in the catalog compiles actually calls this directly: it looks like in multithreaded catalog compiles would just rely on the behavior of `[ ]` to create the thread local settings object